distributed/worker.py

```diff
@@ -1309,6 +1315,14 @@ def update_data(self, comm=None, data=None, report=True, serializers=None):
     async def delete_data(self, comm=None, keys=None, report=True):
         if keys:
             for key in list(keys):
+                if key in self.task_refs:
+                    self.task_refs[key] -= 1
+                    if self.task_refs[key] > 0:
+                        keys.remove(key)
```

Comment:
I think `keys` is a tuple here so you may need to do something else.  Maybe we don't need to remove the key at all ?

Reply:
I was trying to remove it because of the `self.scheduler.remove_keys` call further down.

It seems to be a list so `remove()` should work fine.

```
type(keys)
<class 'list'>
```

Reply:
Hmm you are right. When running Corey's example I get an issue with this being a tuple. Perhaps different calls pass different things?